src/libs/ajax/data-table-providers/WDSDataTableProvider.ts

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash/fp'
+import { Ajax } from 'src/libs/ajax'
+import { DataTableFeatures, DataTableProvider, DeleteTableFn, DownloadTsvFn, EntityMetadata, EntityQueryOptions, EntityQueryResponse, GetMetadataFn, GetPageFn } from 'src/libs/ajax/data-table-providers/DataTableProvider'
+
+// interface definitions for WDS payload responses
+interface AttributeSchema {
+  name: string,
+  datatype: string,
+  relatesTo?: string
+}
+
+export interface RecordTypeSchema {
+  name: string,
+  count: number,
+  attributes: AttributeSchema[]
+}
+
+interface SearchRequest {
+  offset: number,
+  limit: number,
+  sort: 'asc' | 'ASC' | 'desc' | 'DESC',
```

Comment:
hmmm... stuff like this casing variation makes me want to recommend that things like this get "scrubbed" and normalized in some way in the data/provider layer of the UI codebase....